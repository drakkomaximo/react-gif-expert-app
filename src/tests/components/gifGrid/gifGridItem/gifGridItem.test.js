
import { shallow } from 'enzyme'
import GifGridItem from '../../../../components/gifGrid/gifGridItem'

describe('Pruebas de <GifGridItem />', () => {
    let wrapper
    const title = 'prueba'
    const url = 'https://i.pinimg.com/236x/32/d8/f2/32d8f2a21522aa45fc755ee378e45e41.jpg'

    beforeEach(() => {
        wrapper = shallow(<GifGridItem title={title} url={url} />)
    })

    test('debe de mostrar <GifGridItem /> correctamente', () => {

        expect(wrapper).toMatchSnapshot()
    })

    test('debe de tener un parrafo con el titulo', () => {
        const p = wrapper.find('p')
        expect(p.text().trim()).toBe(title)
    })

    test('debe de tener la imagen igual al url y alt de los props', () => {
        const img = wrapper.find('img')
        expect(img.prop('src')).toBe(url)
        expect(img.prop('alt')).toBe(title)
    })

    test('debe tener la clase animate__fadeIn', () => {
        const div = wrapper.find('div')
        const className = div.prop('className')
        expect(className.includes('animate__fadeIn')).toBe(true)
        /* expect( div.prop('className') ).toContain( 'animate__fadeIn' ) */
    })
})
