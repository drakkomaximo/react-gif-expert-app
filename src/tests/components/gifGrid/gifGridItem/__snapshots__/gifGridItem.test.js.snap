// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de <GifGridItem /> debe de mostrar <GifGridItem /> correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="prueba"
    src="https://i.pinimg.com/236x/32/d8/f2/32d8f2a21522aa45fc755ee378e45e41.jpg"
  />
  <p>
     
    prueba
  </p>
</div>
`;
